src/Microsoft.Health.Dicom.Core/Features/Validation/PersonNameValidation.cs

```diff
@@ -27,23 +29,34 @@ public override void Validate(DicomElement dicomElement)
             var groups = value.Split('=');
             if (groups.Length > 3)
             {
-                throw new DicomElementValidationException(name, DicomVR.PN, DicomCoreResource.ValueExceedsAllowedGroups, value);
+                throw new DicomElementValidationException(new PersonNameExceedMaxGroupsError(name, value));
             }
 
             foreach (var group in groups)
             {
-                ElementMaxLengthValidation.Validate(group, 64, $"{name} Group", dicomElement.ValueRepresentation);
+                try
+                {
+                    ElementMaxLengthValidation.Validate(group, 64, name, dicomElement.ValueRepresentation);
+                }
+                catch (DicomElementValidationException ex)
```

Comment:
nit: you can move this predicate up into the catch expression:

```suggestion
                catch (DicomElementValidationException ex) when (ex.Error is ExceedMaxLengthError)
```